cuckoo/common/search.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2017 Cuckoo Foundation.
+# This file is part of Cuckoo Sandbox - http://www.cuckoosandbox.org
+# See the file 'docs/LICENSE' for copying permission.
+
+import re
+from cuckoo.common.elastic import elastic
+from cuckoo.common.mongo import mongo
+
+
+class Search(object):
+    """
+    Search class. Decides which DB abstract layer to use,
+    ElasticSearch or MongoDB.
+    """
+
+    def __init__(self):
+        self.enabled = (elastic.enabled or mongo.enabled)
+
+        self.mongo = mongo
+        self.elastic = elastic
+        return
+
+    def find(self, term, value):
```

Comment:
I'm certainly missing the `magic guesser` part where the `search` module tries to determine the type of search query it's looking at. E.g., when a user enters an IP address, he or she should not be required to also provide the `term` as that could easily be deducted from the search query itself. So a method (with unittests) to do exactly that would go a long way.
This would also be a good start into not having Mongo / Elastic do type guessing themselves (e.g., currently Mongo is checking if the search term is a hash - this logic should be moved into the search query and, if it's a hash, that term should be provided to the mongo class straightaway, simplifying its logic).